sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/ShufflePartitionsUtil.scala

```diff
@@ -47,9 +47,7 @@ object ShufflePartitionsUtil extends Logging {
       advisoryTargetSize: Long,
```

Comment:
are you sure all the assertions in this file should be removed? Which assertion error did you hit?

Reply:
We met the [assertion](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/ShufflePartitionsUtil.scala#L131) inside of `ShufflePartitionsUtil.coalescePartitionsWithSkew`.

> are you sure all the assertions in this file should be removed?

Compared with assertion error, maybe it is better to abort the coalesce operation and continue running the query execution?

Reply:
some assertions can help us find bugs, we shouldn't blindly remove all assertions.

Reply:
Or maybe we could just add assertion for testing?

Reply:
> Or maybe we could just add assertion for testing?

Are you going to do this for all assertions in Spark?